pulsar-client/src/main/java/org/apache/pulsar/client/api/ConsumerConfiguration.java

```diff
@@ -270,6 +272,23 @@ public void setPriorityLevel(int priorityLevel) {
         this.priorityLevel = priorityLevel;
     }
 
+    public boolean getReadCompacted() {
+        return readCompacted;
+    }
+
+    /**
+     * If enabled, the consumer will read messages from the compacted topic rather than reading the full message
+     * backlog of the topic. This means that, if the topic has been compacted, the consumer will only see the latest
+     * value for each key in the topic, up until the point in the topic message backlog that has been compacted.
+     * Beyond that point, the messages will be sent as normal.
+     *
+     * @param readCompacted whether to read from the compacted topic
+     */
+    public ConsumerConfiguration setReadCompacted(boolean readCompacted) {
```

Comment:
if this is controlled at the consumer side (not at subscription level), how does it work (or what application would expect) for shared subscription vs failover subscription?

Reply:
In what I have now, it only works with subs with a single active consumer [1]. I don't see why it couldn't work with multiple active consumers though, the mechanism for reading should be fairly similar.

[1] https://github.com/ivankelly/incubator-pulsar/blob/54964897c91ec972d9ac3fbbbd096e41e028cec6/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherSingleActiveConsumer.java#L300

Reply:
yes, I know that from reviewing your code. However that question I am asking is from user perspective, when user sees this setting in ConsumerConfiguration, they need to know what behavior that should expect. javadoc should be updated then. Also it would be good to fail the subscribe request if shared subscription is specified and readCompacted is configured.

Reply:
Updated